### ISSUE TITLE
RPC functions have mandatory arguments

### ISSUE THREAD

**Issue by KB1RD** [KB1RD/rpcchannel#6]
RPC functions are called with the RPC channel and an array of wildcards, then normal arguments. When making functions that are designed to be used by other parts of the same program, this is awkward since other parts of the program would have to leave the RPC channel argument `undefined` and pass fake wildcards. This means that I usually end up making a real function, then a second RPC function. This really defeats the purpose of those fancy decorators. There instead could be a decorator that functions to re-map RPC function arguments onto normal function arguments. Here's a code snippet that I have floating around:
```typescript
function remapArguments(mapping: ('pass' | 'drop' | 'expand')[]) {
  return function (
    // eslint-disable-next-line
    target: any,
    propertyKey: string,
    descriptor: PropertyDescriptor
  ): PropertyDescriptor {
    const func = descriptor.value
    if (typeof func !== 'function') {
      throw new TypeError('Cannot remap arguments for non-function')
    }
    // TODO: Replace .remapped with a symbol
    // eslint-disable-next-line
    descriptor.value.remapped = function (...args: any[]) {
      const it = mapping[Symbol.iterator]()
      let value: string | undefined = undefined
      return func.apply(this, args.flatMap((v) => {
        if (!value) {
          ;({ value } = it.next())
        }
        switch (value) {
          default:
          case 'pass':
            value = undefined
            return [v]
          case 'drop':
            value = undefined
            return []
          case 'expand':
            // eslint-disable-next-line
            const array: any[] = []
            if (v[Symbol.iterator]) {
              throw new TypeError('Attempted to expand non-iterable')
            }
            const it = v[Symbol.iterator]()
            do {
              const r = it.next()
              if (r.done) {
                throw new TypeError('Expand reached end of array')
              }
              array.push(r.value)
            } while (({ value } = it.next()).value === 'expand')
            return array
        }
      }))
    }
    return descriptor
  }
}
```

**Comment by KB1RD** [comment]
Oh, right: depending on the order that you do this in, this could fix #5

**Comment by KB1RD** [comment]
Fixed in `v0.2.0`
